### ISSUE TITLE
WCAG Techniques H58 wording is not consistent with BCP 47 which does not talk about "primary code"

### ISSUE THREAD

**Issue by cpandhi** [w3c/wcag#1028]
https://www.w3.org/WAI/WCAG21/Techniques/html/H58 states:

Check that the value of the lang attribute conforms to BCP 47: Tags for the Identification of Languages or its successor. 
Verification against the language subtag registry isn't strictly noted by WCAG.
Consistent naming  is useful to avoid confusion.

**Comment by awkawk** [comment]
BCP 47 talks about "primary subtag" - is that what you mean being different? Can you suggest specific changes in the text of H58 that would make sense to you?

**Comment by WilcoFiers** [comment]
Yeah, there are two problems here. Firstly it the technique suggests that lang attributes have to conform to BCP 47. I'd certainly recommend it, but this isn't necessary. If you have something like lang="en-united-states", that will work in all major AT. Browsers / AT (as best I can tell) aren't parsing lang tags according to BCP47, they're using something much more similar to the css `:lang()` pseudo class selector, which is essentially just a "grab the bit before the first dash (`-`)".

The second thing, where it say "Check that the language code matches the language of the content it applies to." The wording is a little odd here, and a little too strict in my opinion. If you have "en-US" on a page with British English spelling, this rule technique isn't met. I realise this is a sufficient technique and they're not required, but this gives the impression that this is required.

Lastly, the terminology in the technique is incorrect. It's "language tag" not "language code" (the whole thing), or "primary language subtag" not "primary code" if you're talking about just the "en" bit in "en-US".

**Comment by awkawk** [comment]
> Yeah, there are two problems here. Firstly it the technique suggests that lang attributes have to conform to BCP 47. I'd certainly recommend it, but this isn't necessary.

I'm not sure that I would say that this is a problem. To satisfy this _technique_ the lang attributes need to conform to BCP 47, but you can still pass the _success criterion_ in other ways. 

Your point about en-US being used on a page with "colour" is fair, but probably not something that most people would say rises to a level that they would describe as a failure. Still, it is an error and it would mean that the content doesn't meet this technique, but would still likely meet the SC since the TTS British and American English synthesizers probably recognize both forms of word variants.

As far as the point about the technique needing to use "tag" instead of "code", it seems that the HTML spec uses "tag":

> 3.2.6.2 The lang and xml:lang attributes
> The lang attribute (in no namespace) specifies the primary language for the element's contents and for any of the element's attributes that contain text. Its value must be a valid BCP 47 language tag, or the empty string

However, the ISO 639-1 spec that is referenced from BCP 47 uses "code":

> ISO 639 provides two language codes, one as a two-letter code (ISO 639-1) and another as a three-letter code (ISO 639-2) for the representation of names of languages.

If someone wants to write up specific changes for the WG to review I'm sure that we can find text that addresses these issues, but as these issues probably aren't causing measurable confusion I imagine that this will be a ways down in the queue.
